python/ray/_private/runtime_env/packaging.py

```diff
@@ -227,19 +227,43 @@ def match(p: Path):
 
 
 def _store_package_in_gcs(
-    pkg_uri: str, data: bytes, logger: Optional[logging.Logger] = default_logger
+    pkg_uri: str,
+    data: bytes,
+    logger: Optional[logging.Logger] = default_logger,
 ) -> int:
+    """Stores package data in the Global Control Store (GCS).
+
+    Args:
+        pkg_uri (str): The GCS key to store the data in.
+        data (bytes): The serialized package's bytes to store in the GCS.
+        logger (Optional[logging.Logger]): The logger used by this function.
+
+    Return:
+        int: Size of data
+
+    Raises:
+        RuntimeError: If the upload to the GCS fails.
+        ValueError: If the data's size exceeds GCS_STORAGE_MAX_SIZE.
+    """
+
     file_size = len(data)
     size_str = _mib_string(file_size)
     if len(data) >= GCS_STORAGE_MAX_SIZE:
-        raise RuntimeError(
+        raise ValueError(
             f"Package size ({size_str}) exceeds the maximum size of "
             f"{_mib_string(GCS_STORAGE_MAX_SIZE)}. You can exclude large "
             "files using the 'excludes' option to the runtime_env."
         )
 
     logger.info(f"Pushing file package '{pkg_uri}' ({size_str}) to Ray cluster...")
-    _internal_kv_put(pkg_uri, data)
+    try:
+        _internal_kv_put(pkg_uri, data)
+    except RuntimeError as e:
```

Comment:
Do you know what the typical failure mode for the `_internal_kv` Python methods are?  Do they hang, or throw an exception?  Would the exception always fall under `RuntimeError`?  @rkooo567 maybe you know?

Reply:
Good questions– I'm not sure about the failure mode.

I checked for a `RuntimeError` because of [`internal_kv_put()`'s error handling](https://github.com/ray-project/ray/blob/f91a134dc6d1bba8cd730cbcf4f9d4b629a08df9/python/ray/_private/gcs_utils.py#L196-L202). Perhaps [`reply = self._kv_stub.InternalKVPut(req)`](https://github.com/ray-project/ray/blob/f91a134dc6d1bba8cd730cbcf4f9d4b629a08df9/python/ray/_private/gcs_utils.py#L195) might raise a different error, but I'm not sure what. @rkooo567 do you know?

Reply:
Gotcha, makes sense!

Reply:
I think it can raise gRPC related errors in some scenarios. Why don't we catch all exceptions instead?